packages/babel-parser/src/parser/statement.js

```diff
@@ -2042,6 +2061,52 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (!this.eat(tt._with)) {
+      if (this.hasPlugin("moduleAttributes")) return [];
+      return null;
+    }
+    this.expectPlugin("moduleAttributes");
+    const attrs = [];
+    const attributeMap = {};
+    do {
+      // we are trying to parse a node which has the following syntax
+      // with type: "json"
+      // [with -> keyword], [type -> Identifier], [":" -> token for colon], ["json" -> StringLiteral]
+      const node = this.startNode();
+      node.key = this.parseIdentifier(true);
+
+      // check if we already have an entry for an attribute
+      // if a duplicate entry found, throw an error
+      if (attributeMap[node.key.name]) {
+        throw this.raise(
```

Comment:
This error is easily recoverable: you can remove `throw`.

Reply:
Noted 👍

Reply:
@nicolo-ribaudo I am not sure why it's showing this conversation here. But to re-confirm should we remove throw here as well?

In this condition we are checking if attribute is anything other than type

Reply:
Yes, let's remove it. `throw` is only needed when it would be impossible to continue paring after the error.